static/src/js/components/RevisionList/RevisionList.jsx

```diff
@@ -0,0 +1,251 @@
+}
+
+/**
+ * Takes a type from the url and returns a humanized singular or plural version
+ * @param {String} type The type from the url.
+ * @param {Boolean} [plural] A boolean that determines whether or not the string should be plural
+ */
+const getHumanizedNameFromTypeParam = (type) => {
+  const humanizedName = typeParamToHumanizedNameMap[type]
+
+  return humanizedName
+}
+
+/**
+ * Renders a `RevisionList` component
+ *
+ * @component
+ * @example <caption>Renders a `RevisionList` component</caption>
+ * return (
+ *   <RevisionList />
+ * )
+ */
+const RevisionList = ({ limit }) => {
+  const { conceptId, type } = useParams()
+  const [searchParams, setSearchParams] = useSearchParams()
+  const versions = searchParams.get('versions')
+  const formattedType = capitalize(type)
+  const activePage = parseInt(searchParams.get('page'), 10) || 1
+  const offset = (activePage - 1) * limit
+
+  const { revisions, error, loading } = useRevisionsQuery({
+    conceptId,
+    type: formattedType,
+    limit,
+    offset,
+    sortKey: '-revisionDate'
+  })
+
+  const setPage = (nextPage) => {
+    setSearchParams((currentParams) => {
+      currentParams.set('page', nextPage)
+
+      return {
+        ...Object.fromEntries(currentParams)
+      }
+    })
+  }
+
+  const { count } = revisions
+  const { items = [] } = revisions
+
+  const buildDescriptionCell = useCallback((cellData) => {
+    // The reason I cannot use count, is because it resets to undefined
+    // upon refreshing the page. Even when I put count in as a dependency.
+    // using async and await here doesn't seem to work with the useCallback function.
+    // const convertedCellData = parseInt(cellData, 10)
+    // const isPublishedVersion = (convertedCellData === count)
+    // console.log("🚀 ~ buildDescriptionCell ~ count:", count)
+    // console.log("🚀 ~ buildDescriptionCell ~ convertedcellData:", convertedCellData)
+
+    const isPublishedVersion = (cellData === versions)
+
+    return (
+      <EllipsisLink to={`/${type}/${conceptId}/${cellData}`} inline>
+        {[cellData, ' - ', (isPublishedVersion ? 'Published' : 'Revision')].join('')}
+      </EllipsisLink>
+    )
+  }, [])
+
+  // To be done after GQL-32
+  const buildActionCell = useCallback((cellData) => {
+    const isPublishedVersion = (cellData === versions)
+
+    return (
+      isPublishedVersion
+        ? ' '
+        : 'Revert to this version'
+    )
+  }, [])
+
+  const getColumnState = () => [
+    {
+      dataKey: 'revisionId',
+      title: 'Description',
+      className: 'col-auto',
+      dataAccessorFn: buildDescriptionCell
+    },
+    {
+      dataKey: 'revisionDate',
+      title: 'Revision Date',
+      className: 'col-auto',
+      dataAccessorFn: (cellData) => cellData.split('T')[0]
+    },
+    {
+      dataKey: 'userId',
+      title: 'Action by',
+      className: 'col-auto'
+    },
+    {
+      // Change dataKey to 'revisionId' then fix unique key issue (see line 116)
```

Comment:
I'm not sure if I'll be working on this file when we come back to it in sprint two so it's a note for the next dev. Line 98 will need to take in 'revisionId' instead of 'actions' and this causes a unique key I'd need to dig into to figure out, personally. The keys are being set on line 116.